From: Li Dongyang <dongyangli@ddn.com>
Date: Wed, 15 Jan 2025 15:36:05 +1100
Subject: [PATCH] misc: fix --without-libarchive build failure
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

configure --without-libarchive triggers the error:

./../misc/create_inode_libarchive.c: In function ‘__populate_fs_from_tar’:
./../misc/create_inode_libarchive.c:25:34: error: parameter name omitted
 errcode_t __populate_fs_from_tar(ext2_filsys, ext2_ino_t, const char *,
                                  ^~~~~~~~~~~
./../misc/create_inode_libarchive.c:25:47: error: parameter name omitted
 errcode_t __populate_fs_from_tar(ext2_filsys, ext2_ino_t, const char *,
                                               ^~~~~~~~~~
./../misc/create_inode_libarchive.c:25:59: error: parameter name omitted
 errcode_t __populate_fs_from_tar(ext2_filsys, ext2_ino_t, const char *,
                                                           ^~~~~~~~~~~~
./../misc/create_inode_libarchive.c:26:34: error: parameter name omitted
                                  ext2_ino_t, struct hdlinks_s *,
                                  ^~~~~~~~~~
./../misc/create_inode_libarchive.c:26:46: error: parameter name omitted
                                  ext2_ino_t, struct hdlinks_s *,
                                              ^~~~~~~~~~~~~~~~~~
./../misc/create_inode_libarchive.c:27:34: error: parameter name omitted
                                  struct file_info *,
                                  ^~~~~~~~~~~~~~~~~~
./../misc/create_inode_libarchive.c:28:34: error: parameter name omitted
                                  struct fs_ops_callbacks *) {
                                  ^~~~~~~~~~~~~~~~~~~~~~~~~

Fixes: ecfd4dd1217a ("Decouple --without-libarchive and HAVE_ARCHIVE_H")

Signed-off-by: Li Dongyang <dongyangli@ddn.com>
---
 misc/create_inode_libarchive.c | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/misc/create_inode_libarchive.c b/misc/create_inode_libarchive.c
index 9c8e53e4674e..f89ac1e4c343 100644
--- a/misc/create_inode_libarchive.c
+++ b/misc/create_inode_libarchive.c
@@ -22,10 +22,11 @@
 
 /* If ./configure was run with --without-libarchive, then only
  * __populate_fs_from_tar() remains in this file and will return an error. */
-errcode_t __populate_fs_from_tar(ext2_filsys, ext2_ino_t, const char *,
-                                 ext2_ino_t, struct hdlinks_s *,
-                                 struct file_info *,
-                                 struct fs_ops_callbacks *) {
+errcode_t __populate_fs_from_tar(ext2_filsys fs, ext2_ino_t root_ino,
+				 const char *source_tar, ext2_ino_t root,
+				 struct hdlinks_s *hdlinks,
+				 struct file_info *target,
+				 struct fs_ops_callbacks *fs_callbacks) {
   com_err(__func__, 0,
           _("you need to compile e2fsprogs without --without-libarchive"
             "be able to process tarballs"));
